### PR TITLE
Monitoring library in DPL

### PR DESCRIPTION
**Pull request by awegrzyn**

### CODE REVIEW

**Comment by awegrzyn** [comment]
@ktf Could you let me know what else need to be done to merge this?

File: Framework/Core/src/DataProcessingDevice.cxx [hunk 1]
```diff
@@ -95,6 +98,7 @@ DataProcessingDevice::HandleData(FairMQParts &iParts, int /*index*/) {
   // does not need to know about the whole class state, but I can 
   // fine grain control what is exposed at each state.
   auto &metricsService = mServiceRegistry.get<MetricsService>();
+  auto& monitoringService = mServiceRegistry.get<Monitoring>();
```
Comment (ktf) [inline Framework/Core/src/DataProcessingDevice.cxx]: For the record, as we discussed, I think we should replace completely the `MetricsService` with` Monitoring`.

File: Framework/Core/test/DummyMetricsService.h [hunk 2]
```diff
@@ -1,28 +0,0 @@
-// Copyright CERN and copyright holders of ALICE O2. This software is
```
Comment (ktf) [inline Framework/Core/test/DummyMetricsService.h]: Can we have a "no-op" implementation also for the `Monitoring` interface? It comes handy for tests.

Comment (awegrzyn) [reply Framework/Core/test/DummyMetricsService.h]: You mean a backend that does nothing? If so, of couse.

Comment (ktf) [reply Framework/Core/test/DummyMetricsService.h]: yes, that's what I meant.

Comment (awegrzyn) [reply Framework/Core/test/DummyMetricsService.h]: `MonitoringFactory::Get("no-op://")` is available from v1.7.0

**Comment by awegrzyn** [comment]
Is it OK to merge it?

**Comment by ktf** [comment]
I had no time to test this last week for personal reasons. Did you verify the plots in the debug GUI are the same?

**Comment by awegrzyn** [comment]
Nope, how can I run debug GUI? Where's the code responsible to passing monitoring values to the GUI?

**Comment by ktf** [comment]
The GUI should pop up whenever you run any of the examples, e.g. `o2DiamondWorkflow`. The code which does the parsing of the metrics can be found in `DeviceMetricInfo.{cxx,h}`.

**Comment by awegrzyn** [comment]
Currently I'm stuck as on my Mac I can't compile O2: https://github.com/AliceO2Group/AliceO2/issues/1077
and on remote CC7 GLFW fails due to X11 forwarding.

**Comment by awegrzyn** [comment]
All right,
I have a small problem with ServiceRegistry. I've been getting "segmentation violation" as `serviceRegistry.get<Monitoring>();` somehow calls `Monitoring(const Monitoring&)` constructor which is explicitly deleted.
I thought ServiceRegistry just keeps a pointer to a service, it's not the case?

**Comment by awegrzyn** [comment]
Okay I found the issue. It was `.get()` from `unique_ptr` not from `serviceRegistry`!

**Comment by ktf** [comment]
The code in question is at:

https://github.com/AliceO2Group/AliceO2/blob/7d1b1bd058d4b34dcc23fe9be1ffc0b40161655c/Framework/Core/include/Framework/ServiceRegistry.h#L47

and I do not see where it could call the copy constructor. Can you get a better stacktrace of where that happens?

**Comment by awegrzyn** [comment]
Okay, graphs in GUI work with the last changes. Could you please verify?

**Comment by awegrzyn** [comment]
Okay, I need to adapt tests. Working on it...

**Comment by awegrzyn** [comment]
@dberzano I'm not sure why `build/o2checkcode/o2` is failing? From the logs everything looks ok.

**Comment by ktf** [comment]
We saw the same with @matthiasrichter the other day. We think it's the ninja support getting in the way.

**Comment by sawenzel** [comment]
This test has been constantly failing for all PRs for the last days. Is someone looking into this?

**Comment by awegrzyn** [comment]
Ok in such case I think it's ready to be merged.

**Comment by sawenzel** [comment]
clang-format is also still failing (and we might use this occassion to adiabatically correct the code with respect to the formatting)

**Comment by awegrzyn** [comment]
@sawenzel Is there a way to apply this changes automatically?

**Comment by matthiasrichter** [comment]
I did a quick test yesterday to check whether this could be related to https://github.com/alisw/alidist/pull/1136 but it does not seem the case. I can reproduce the problem locally, but reverting to a revision before https://github.com/alisw/alidist/pull/1136 did not help.

Unfortunately I won't have time to look more into this the next days.

**Comment by matthiasrichter** [comment]
@awegrzyn, see instructions in the main README https://github.com/AliceO2Group/AliceO2#formatting

**Comment by ktf** [comment]
I tried this over the weekend, but it looks like the parser is not matching the new format. Is that supposed to happen?

**Comment by awegrzyn** [comment]
Hi,  I've adapted `test_DeviceMetricsInfo.cxx` which successfully passes and verified that plots in GUI update.
Could you let me know what does not work and how I can reproduce it?

**Comment by ktf** [comment]
My bad. I forgot to recompile.

**Comment by ktf** [comment]
Tested locally, works fine. Error unrelated. I will have a few comments in general about the Monitoring API, but that can come later.

**Pull request merged**